packages/yourloops/test/integration/pages/patient-dashboard/patient-dashboard-for-patient.spec.tsx

```diff
@@ -48,3 +56,54 @@ describe('Patient dashboard for HCP', () => {
     expect(secondaryHeader).toHaveTextContent('DashboardDailyTrendsGenerate report')
   })
 })
+
+describe('Team menu for patient', () => {
+  beforeAll(() => {
+    mockPatientLogin(monitoredPatientAsTeamMember)
+    jest.spyOn(TeamAPI, 'getTeams').mockResolvedValue([teamOne])
+    jest.spyOn(TeamAPI, 'joinTeam').mockResolvedValue()
+    jest.spyOn(TeamAPI, 'getTeamFromCode').mockResolvedValue(iTeamOne)
+  })
+
+  it('should close the dialog after clicking the join care team button with success message', async () => {
+    await act(async () => {
+      renderPage('/')
+    })
+
+    const badgeTeamMenu = screen.getByLabelText('Open team menu')
+    expect(badgeTeamMenu).toHaveTextContent('1')
+    await userEvent.click(badgeTeamMenu)
+    const buttonJoinTeam = screen.getByTestId('team-menu-teams-link')
+    await userEvent.click(buttonJoinTeam)
+    const dialog = screen.getByRole('dialog')
+    expect(dialog).toHaveTextContent('Join the care team Please enter the team code (9 digits)CancelContinue')
+    const addTeamButton = within(dialog).getByRole('button', { name: 'Continue' })
+    const inputCode = within(dialog).getByRole('textbox')
+    expect(addTeamButton).toBeDisabled()
+    await userEvent.type(inputCode, '263381988')
+    expect(addTeamButton).toBeEnabled()
+    await userEvent.click(addTeamButton)
+    expect(TeamAPI.getTeamFromCode).toHaveBeenCalledWith('263381988')
+
+    const dialogPrivacy = screen.getByRole('dialog')
+    const addCareTeamButton = within(dialogPrivacy).getByRole('button', { name: 'Join team' })
+    const checkPolicy = within(dialogPrivacy).getByRole('checkbox')
+    expect(dialogPrivacy).toHaveTextContent('Share your data with a care teamYou ' +
+      'are about to share you data with' +
+      'iTeamOne6 rue des champs75000Paris679517388' +
+      'Please verify that the above details match the information provided by your healthcare professional before accepting the invitation.' +
+      'By accepting this invitation, I recognize this team as my care team and consent to share my personal data with all its members, who are authorized healthcare professionals registered on YourLoops. ' +
+      'I acknowledge that I can revoke this access at any time.Read our Privacy Policy for more information.CancelJoin team')
+    expect(addCareTeamButton).toBeDisabled()
+    await userEvent.click(checkPolicy)
+    expect(addCareTeamButton).toBeEnabled()
+    await userEvent.click(addCareTeamButton)
+    expect(TeamAPI.joinTeam).toHaveBeenCalledWith(iTeamOne.id, unmonitoredPatientId)
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+    expect(screen.getByText('Your care team has now access to your data.')).toBeVisible()
+
+    closeDialogJoinTeam()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏Missing `await` before the 3 following function calls